### PR TITLE
Total time to resolution & total time over sla per jira issue

### PR DESCRIPTION
**Pull request by gabezurita**
Good afternoon @bherrup, 

This update to the Jira Reporting Gem will be used by Ed to track and save the following on individual Tech Support Jira issues:

- a Jira issue's total time to resolution
- a Jira issue's total time over sla

In aggregate, these fields will be used to generate stats and track the above for all Tech Support Jira issues and for individual tech teams. The plan is to graph and report on this data in the monthly report, as well as the KPI weekly report.

This [jira test issue](https://optoro.atlassian.net/browse/TS-7819) has the relevant Jira fields with the information that's being saved under the "Tracking Info" tab -- see screenshot:

image

If you have any suggestions on how I can improve any portion of this implementation, I would appreciate it!

Thanks,
Gabo

### CODE REVIEW

File: bin/average_time_to_close [hunk 1]
```diff
@@ -91,22 +91,22 @@ puts "Total total_ts_cumulative_delay: #{total_ts_cumulative_delay} hours"
...
-    report[:sla][:p2][:avg_hit_rate].to_s + '%',
-    report[:sla][:p2][:avg_time_to_close].to_s,
-    report[:sla][:p3][:avg_hit_rate].to_s + '%',
-    report[:sla][:p3][:avg_time_to_close].to_s,
-    report[:sla][:p4][:avg_hit_rate].to_s + '%',
-    report[:sla][:p4][:avg_time_to_close].to_s
-  ].join("\n")
-end
+# def copy_paste_string(report)
+#   [
```
Comment (bherrup) [inline bin/average_time_to_close]: why not just delete this?

File: bin/average_time_to_close [hunk 2]
```diff
@@ -35,9 +35,9 @@ closed_ts_issues = JiraIssue.find(closed_ticket_query)
 sla_counted_issues = closed_ts_issues.reject { |issue| issue.sla_due_time.nil? }
 
 total_closed_ts_issues = closed_ts_issues.count
-total_time_to_resolution = closed_ts_issues.inject(0){|sum, issue| sum + issue.time_to_resolution_in_hours }
+total_time_to_resolution = closed_ts_issues.inject(0){|sum, issue| sum + issue.sum_total_time_to_resolution_in_hours }
 average_time_to_resolution = (total_time_to_resolution / total_closed_ts_issues).round(2)
-total_ts_cumulative_delay = sla_counted_issues.inject(0){|sum, issue| sum + issue.sum_total_time_over_sla }.round(2)
+total_ts_cumulative_delay = sla_counted_issues.inject(0){|sum, issue| sum + issue.sum_total_time_over_sla_in_hours }.round(2)
```
Comment (bherrup) [inline bin/average_time_to_close]: where is this defined?

File: bin/average_time_to_close (hunk 1 above)
Comment (gabezurita) [reply bin/average_time_to_close]: You are totally right, I should just delete this. It's functionality that I might need eventually that's not fully hashed out, but the YAGNA practice dictatest: "Always implement things when you actually need them, never when you just foresee that you need them."  http://wiki.c2.com/?YouArentGonnaNeedIt

Thanks for the feedback Brad! I'll delete this portion and make another PR.

File: bin/average_time_to_close (hunk 2 above)
Comment (gabezurita) [reply bin/average_time_to_close]: Woops. I renamed the method and did not update this instance of it -- thanks for catching that!

This script was generally unfinished and really beyond the scope of this PR, but I thought I might as well get started on it. I'll make a few changes to it and make another PR.

**Comment by gabezurita** [comment]
Good afternoon @bherrup,

Thanks for all the feedback and guidance -- I've made the suggested changes. Please let me know if I can improve anything else!

Thank you,
Gabo

**Pull request merged**